lib/internal/process/main_thread_only.js

```diff
@@ -29,12 +31,23 @@ function setupStdio() {
 // Non-POSIX platforms like Windows don't have certain methods.
 // Workers also lack these methods since they change process-global state.
 function setupProcessMethods(_chdir, _umask, _initgroups, _setegid,
-                             _seteuid, _setgid, _setuid, _setgroups) {
+                             _seteuid, _nice, _setgid, _setuid,
+                             _setgroups) {
+
   if (_setgid !== undefined) {
     setupPosixMethods(_initgroups, _setegid, _seteuid,
                       _setgid, _setuid, _setgroups);
   }
 
+  process.nice = function nice(inc) {
+    if (process.platform === 'win32') {
```

Comment:
Perhaps change this so the platform check is only run once when this file is evaluated, since the process platform will not change during runtime. Example:

```js
if (process.platform === 'win32') {
  process.nice = function nice(inc) {
    throw new ERR_METHOD_NOT_IMPLEMENTED('nice()');
  };
} else {
  process.nice = function nice(inc) {
    validateInc(inc);
    return _nice(inc);
  };
}
```